.github/workflows/develop.yml

```diff
@@ -0,0 +1,62 @@
+
+name: develop pre-release
+
+on:
+  push:
+    branches:
+      - main
+env:
+  GRADLE_OPTS: "-Dorg.gradle.parallel=true -Dorg.gradle.caching=true"
+
+jobs:
+  artifacts:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+    steps:
+      - name: checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      - name: Set up JDK 17
+        uses: actions/setup-java@387ac29b308b003ca37ba93a6cab5eb57c8f5f93
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+      - name: setup gradle
+        uses: gradle/actions/setup-gradle@9e899d11ad247ec76be7a60bc1cf9d3abbb9e7f1
+      - name: assemble release
+        run:
+          ./gradlew assemble
+      - name: upload tarball
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3
+        with:
+          path: 'build/distributions/besu*.tar.gz'
```

Comment:
do we not need the girhub.ref_name in the path?

Reply:
Good call, that will always be `main` so we can replace that with something static.